app/jobs/unsubscribe_from_list_job.rb

```diff
@@ -0,0 +1,17 @@
+class UnsubscribeFromListJob < ApplicationJob
+  queue_as :default
+
+  def perform(subscription_id)
+    @client = Pardot.client
+    subscription = Subscription.find(subscription_id)
+
+    prospect_id = salesforce_to_prospect(subscription.contact.salesforce_id)
+
+    @client.list_memberships.delete(subscription.list.pardot_id, prospect_id)
+    subscription.destroy
```

Comment:
If this fails, what do you want to happen?  Would fail if destroy callbacks on `Subscription` failed.  These may not exist now, but what should the behavior be?  Also the line above has an unresolved and related comment from the last review

Reply:
Missed the unresolved comments since the file was changed - responded to those (the job will fail).
I added some code to recreate the local record if the destroy fails. Still thinking if this is the ideal way to handle this. 🤔